File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.js
/**
 * @param {number[]} plantTime
 * @param {number[]} growTime
 * @return {number}
 */
var earliestFullBloom = function(plantTime, growTime) {
    let restDay = 0;
    let moreNecessaryDays = [];
    let combined = plantTime.map((p, i) => [p, growTime[i]]);
    combined.sort((a, b) => a[1] - b[1]);
    
    for (let [plantDay, growDay] of combined) {
        if (restDay < growDay) {
            moreNecessaryDays.push(growDay - restDay);
        }
        restDay += plantDay;
    }
    return plantTime.reduce((a, b) => a + b, 0) + Math.max(...moreNecessaryDays);
};

File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.py
class Solution:
    def earliestFullBloom(self, plantTime: List[int], growTime: List[int]) -> int:
        rest_day = 0
        more_necessary_days = []
        for plant_day, grow_day in sorted(zip(plantTime, growTime), key=lambda x: x[1]):
            if rest_day < grow_day:
                more_necessary_days.append(grow_day - rest_day)
            rest_day += plant_day
        return sum(plantTime) + max(more_necessary_days)